src/Microsoft.AspNetCore.Blazor/Components/IHandleAfterRender.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Blazor.Components
+{
+    /// <summary>
+    /// Interface implemented by components that receive notification that they have been rendered.
+    /// </summary>
+    public interface IHandleAfterRender
+    {
+        /// <summary>
+        /// Notifies the component that it has been rendered.
+        /// </summary>
+        void OnAfterRender();
```

Comment:
Why did you choose to make the interface synchronous and then have BlazorComponent have an asynchronous overload?

Feels to me more natural to have this interface be asynchronous and just reverse what you do in the implementation.
```csharp
public virtual Task OnAfterRenderAsync()
{
    OnAfterRender();

    return Task.CompletedTask;
}

protected virtual void OnAfterRender() 
{
}
```
We do a similar pattern in filters in MVC. See
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Filters/ActionFilterAttribute.cs

Reply:
That's a good point, and again I agree that would be better. Also again I'll keep it consistent with the rest of the lifecycle methods for this PR and we can do a broader review of them all later.